packages/styled-components/src/createStyled.ts

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable no-continue, no-loop-func, no-cond-assign */
+import type { ElementType } from 'react'
+import { BoxElements } from '@xstyled/core'
+import { string } from '@xstyled/util'
+import { StyleGenerator, StyleGeneratorProps } from '@xstyled/system'
+import scStyled, {
+  DefaultTheme,
+  StyledConfig,
+  ThemedBaseStyledInterface,
+  ThemedStyledFunction,
+} from 'styled-components'
+import { css } from './css'
+
+const getCreateStyle = (
+  baseCreateStyle: ThemedStyledFunction<any, any>,
+  generator?: StyleGenerator,
+) => {
+  const createStyle = (...args: Parameters<typeof css>) =>
+    // @ts-ignore
+    baseCreateStyle`${css(...args)}${generator}`
+  createStyle.attrs = (attrs: Parameters<typeof baseCreateStyle.attrs>[0]) =>
+    getCreateStyle(baseCreateStyle.attrs(attrs), generator)
+  createStyle.withConfig = (config: StyledConfig<any>) =>
+    getCreateStyle(baseCreateStyle.withConfig(config), generator)
+  return createStyle
+}
+
+type BoxStyledTags<TProps extends object> = {
+  [Key in keyof BoxElements]: ThemedStyledFunction<
+    BoxElements[Key],
+    DefaultTheme,
+    TProps
+  >
+}
+
+export interface CreateXStyled<TProps extends object = {}>
+  extends ThemedBaseStyledInterface<DefaultTheme>,
+    BoxStyledTags<TProps> {}
+
+const createShouldForwardProp = (
+  generator: StyleGenerator,
+): ((
+  prop: string | number | symbol,
+  defaultValidatorFn: (prop: string | number | symbol) => boolean,
+  elementToBeCreated?: ElementType,
+) => boolean) => {
+  const propSet = new Set<string>(generator.meta.props)
+  return (
+    prop: string | number | symbol,
+    defaultValidatorFn: (prop: string | number | symbol) => boolean,
+    elementToBeCreated?: ElementType,
+  ) => {
+    if (string(prop) && propSet.has(prop)) {
+      return false
+    }
+    if (typeof elementToBeCreated === 'string') {
+      // We must test elementToBeCreated so we can pass through props for <x.div
+      // as={Component} />. However elementToBeCreated isn't available until
+      // styled-components 5.2.4 or 6, and in the meantime will be undefined.
```

Comment:
FYI this fix was released in styled-components 5.3.0 — both of my guesses were wrong!

Reply:
I will change it too.